packages/affiliates/libs/uniswap/utils.ts

```diff
@@ -0,0 +1,33 @@
+import assert from "assert";
+import { ethers } from "ethers";
+// taken from uniswap subgraph, not sure if correct yet
+const Q192: BigInt = 2n ** 192n;
+export function sqrtPriceX96ToTokenPrices(sqrtPriceX96: BigInt): string[] {
+  const num = BigInt(sqrtPriceX96) * BigInt(sqrtPriceX96);
+  const denom = Q192;
+  const price1 = BigInt(num) / BigInt(denom);
+  const price0 = 1n / price1;
+  return [price0.toString(), price1.toString()];
+}
+
+// check if a value is not null or undefined, useful for numbers which could be 0
+/* eslint-disable-next-line @typescript-eslint/ban-types */
+export function exists(value: any): value is {} {
```

Comment:
what is this syntax `value is {}`?

Reply:
actually spent quite a bit of time researching how to represent a value which is not null and not undefined in typescript. This `{}` i think is the closest representation of this that i found, but for some reason is disallowed by default in eslint. 

from a [stackoverflow](https://stackoverflow.com/questions/60381643/what-does-mean-in-typescript):

> So {} imposes almost no constraints on its values; the only rule is that it can't be null or undefined. It is similar to the type object in this regard, except that object also forbids primitive values, while {} allows them

Reply:
and i just realized you are asking about `value is {}`, i dont remember actually why this had to be there, but i had trouble getting this function to compile without it. will look into it some more

Reply:
Heres why https://stackoverflow.com/questions/40081332/what-does-the-is-keyword-do-in-typescript

Basically im using it as a guard, and typescript wont be able to tell that its guarding againts null or undefined, and will keep erroring unless you use 'is'

Reply:
Very cool!